backend/installGlobalHook.js

```diff
@@ -31,29 +30,17 @@ function installGlobalHook(window: Object) {
           // but might add 2 (PROFILE) in the future.
           return 'development';
         }
-        // The above should cover envification, but we should still make sure
-        // that the bundle code has been uglified.
-        var findFiberCode = toString.call(renderer.findFiberByHostInstance);
-        // Filter out bad results (if that is even possible):
-        if (findFiberCode.indexOf('function') !== 0) {
-          // Hope for the best if we're not sure.
-          return 'production';
-        }
-
-        // By now we know that it's envified--but what if it's not minified?
-        // This can be bad too, as it means DEV code is still there.
-
-        // FIXME: this is fragile!
-        // We should replace this check with check on a specially passed
-        // function. This also doesn't detect lack of dead code elimination
-        // (although this is not a problem since flat bundles).
-        if (findFiberCode.indexOf('getClosestInstanceFromNode') !== -1) {
-          return 'unminified';
-        }
 
-        // We're good.
+        // React 16 uses flat bundles. If we report the bundle as production
+        // version, it means we also minified and envified it ourselves.
         return 'production';
+        // Note: There is still a risk that the CommonJS entry point has not
+        // been envified or uglified. In this case the user would have *both*
+        // development and production bundle, but only the prod one would run.
+        // This would be really bad. We have a separate check for this because
+        // it happens *outside* of the renderer injection. See `checkDCE` below.
       }
+      var toString = Function.prototype.toString;
```

Comment:
Curious why we do this (vs `Function.prototype.toString.call`)?

Reply:
No reason. It used to be used in more places but now it's not.